website/blog/2022-06-30-extract-sql-function.md

```diff
@@ -0,0 +1,85 @@
+---
+title: "EXTRACT SQL function: Why we love it"
+description: "In this post, we’re going to give a deep-dive into the EXTRACT function, how it works, and why we use it. The EXTRACT function allows you to extract a specified date part from a date/time. "
+slug: extract-sql-love-letter
+
+authors: [kira_furuichi]
+
+tags: [SQL Magic]
+hide_table_of_contents: false
+
+date: 2022-06-30
+is_featured: false
+---
+There are so many different date functions in SQL—you have DATEDIFF, [DATEADD](https://docs.getdbt.com/blog/sql-dateadd), DATE_PART, and DATE_TRUNC to name a few. They all have their different use cases and understanding how and when they should be used is a SQL fundamental to get down. Are any of those as easy to use as the EXTRACT function? Well, that debate is for another time…
+
+In this post, we’re going to give a deep dive into the EXTRACT function, how it works, and why we use it.
+
+<!--truncate-->
+
+The EXTRACT function allows you to extract a specified date part from a date/time. For example, if you were to extract the month from the date February 14, 2022, it would return 2 since February is the second month in the year.
+
+> **What is a SQL function?**
+> At a high level, a function takes an input (or multiple inputs) and returns a manipulation of those inputs. Some common SQL functions are COALESCE, LOWER, and DATEDIFF. For example, the `COALESCE` function takes a group of values and returns the first non-null value from that group.
+
+## How to use the EXTRACT function
+
+One of our favorite things about the EXTRACT function is how readable it is. Sometimes you may encounter SQL functions and not immediately understand what the arguments are and what the expected output should be. (We’re looking at you, SPLIT_PART.) The EXTRACT function isn’t like that.
+
+To use the EXTRACT function, you’ll simply specify the date part you want extracted out and the field you want to extract from. You can extract many different date parts, but you’ll most commonly see year, month, week of year, or quarter extracted from a date.
+
+```sql
```

Comment:
can you play around with what this looks like as JS or yaml? The coloring is off since this isn't like technically proper sql syntax.

<img width="387" alt="image" src="https://user-images.githubusercontent.com/18452766/176769904-c49c2e60-55d0-4f84-a96f-933321bd80a6.png">

Reply:
It's fine if it's all one color